rebase: use v0.17.0 of go-ceph library

humblec opened the pull request:
new version of go ceph is available and this commit make use
of the same.
Ref # https://github.com/ceph/go-ceph/releases/tag/v0.17.0

This go-ceph version also bring changes to some APIs which is part of pre release tag of Ceph build quincy.
Enabling the tag `ceph_pre_quincy` for the same reason.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec committed:
rebase: use v1.17.0 of go-ceph library

new version of go ceph is available and this commit make use
of the same.
Ref # https://github.com/ceph/go-ceph/releases/tag/v0.17.0

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

Madhu-1 commented:
@humblec CI is failing can you please check

humblec commented:
The new APIs are causing the failure.. the go-ceph part of these APIs are "preview mode" ,

``` added by https://github.com/ceph/go-ceph/pull/691/files
  internal/cephfs/core/metadata.go:86:12: fsa.RemoveMetadata undefined (type *"github.com/ceph/go-ceph/cephfs/admin".FSAdmin has no field or method RemoveMetadata) (typecheck)
	err = fsa.RemoveMetadata(s.FsName, s.SubvolumeGroup, s.VolID, key)
	          ^
internal/cephfs/core/snapshot_metadata.go:64:12: fsa.SetSnapshotMetadata undefined (type *"github.com/ceph/go-ceph/cephfs/admin".FSAdmin has no field or method SetSnapshotMetadata) (typecheck)
	err = fsa.SetSnapshotMetadata(s.FsName, s.SubvolumeGroup, s.VolID, s.SnapshotID, key, value)
	          ^
internal/cephfs/core/snapshot_metadata.go:83:12: fsa.RemoveSnapshotMetadata undefined (type *"github.com/ceph/go-ceph/cephfs/admin".FSAdmin has no field or method RemoveSnapshotMetadata) (typecheck)
	err = fsa.RemoveSnapshotMetadata(s.FsName, s.SubvolumeGroup, s.VolID, s.SnapshotID, key)

```
Ceph build's `ceph_preview` tag is enabled for our builds already, may be we need a different tag `ceph_pre_{pacific,quincy}, ceph_main` ? 

Cc @nixpanic @pkalever

nixpanic commented:
Looks like the CephFS metadata functions are only valid for pre-Quincy versions? Not sure if Quincy requires other functions, or how it is done there. Maybe @pkalever knows.

pkalever commented:
I think the reason we added those APIs as the preview was that the intended Ceph PRs introducing those metadata commands were still in master and not released in any Ceph release yet. Need to check if they were available in the latest release of Ceph and should move them to stable API sections in go-ceph.

> Looks like the CephFS metadata functions are only valid for pre-Quincy versions? Not sure if Quincy requires other functions, or how it is done there. Maybe @pkalever knows.

Quincy and pre-quincy have the same commands and APIs, the original Quincy PRs are backported to the pacific and octopus.

humblec committed:
build: enable ceph_pre_quincy tag

This commit add ceph_pre_quincy tag to the build. The main
reason being the changes or new APIs exposed in go-ceph version
v0.17.0 is part of this Ceph build tag. Enabling the same for
the reason.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec commented:
As in https://github.com/ceph/ceph-csi/pull/3308#issuecomment-1217773138 , I think one of the tags below should help .
ceph_pre_{qunicy/pacific}
or 
ceph_main

pkalever commented:
> I think the reason we added those APIs as the preview was that the intended Ceph PRs introducing those metadata commands were still in master and not released in any Ceph release yet. Need to check if they were available in the latest release of Ceph and should move them to stable API sections in go-ceph.

Looks like these metadata commands are targeted for Ceph - v18.0.0, which is not out yet!, we still need to wait on this!

humblec commented:
@nixpanic @pkalever as anticipated `ceph_pre_quincy` helps here. ptal .. thanks.

humblec commented:
@Mergifyio rebase

humblec commented:
Tests are stuck , rebasing due to the same reason .

humblec commented:
/retest ci/centos/k8s-e2e-external-storage/1.22

humblec commented:
/retest ci/centos/upgrade-tests-cephfs

humblec commented:
/retest ci/centos/mini-e2e/k8s-1.24

humblec commented:
/retest ci/centos/mini-e2e/k8s-1.24

humblec commented:
/retest ci/centos/mini-e2e-helm/k8s-1.22

Rakshith-R commented:
@mergifyio rebase
